core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -460,16 +462,11 @@ public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throw
     mStatistics = statistics;
     mUri = URI.create(mAlluxioHeader);
 
+    Map<String, Object> uriConfProperties = getConfigurationFromUri(uri);
+
     synchronized (INIT_LOCK) {
-      if (sInitialized) {
-        if (!connectDetailsMatch(mUri, conf)) {
-          LOG.warn(ExceptionMessage.DIFFERENT_MASTER_ADDRESS
-              .getMessage(mUri.getHost() + ":" + mUri.getPort(),
-                  FileSystemContext.get().getMasterAddress()));
-          initializeInternal(uri, conf);
-        }
-      } else {
-        initializeInternal(uri, conf);
+      if (!sInitialized || !connectDetailsMatch(uriConfProperties, conf)) {
+        initializeInternal(uriConfProperties, conf);
```

Comment:
Maybe adding back LOGGING on `DIFFERENT_CONNECTION_DETAILS` 
when `sInitialized` is true? I 
I think this is a useful logging though even you mentioned that we have logged connection details in other places. It may not be easy to find the previous connection details with only that.

Reply:
The newConnectDetails is logged in the `InitializeInternal` and if outside the` connectDetailsMatch` utils method, we have to compute again to get it.
@apc999 How about log the old connect details here and log the new connect details in the `InitializeInternal`.

Reply:
Add back the logging here to tell users the old connection details.

Reply:
@LuQQiu I don't see this `DIFFERENT_CONNECTION_DETAILS` is used